nixos/install-tools: Add manpages to packages instead of seperating them

doronbehar opened the pull request:
###### Motivation for this change


###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

doronbehar committed:
nixos-rebuild: Include manual page in the package

doronbehar committed:
nixos-option: Include manual page in the package

doronbehar committed:
nixos/install-tools: Add manpages to packages instead of seperating them

Since each such `nixos-*` tool has it's own derivation, exposed in pkgs,
There is no point in separating the manuals from the packages. If
someone wishes to have the tools without the manuals, they can use
meta.outputsToInstall to disable the installation of the manpages of
these packages. This Fixes #244450.

roberth commented on pkgs/tools/nix/nixos-install-tools/default.nix:
```diff
@@ -21,7 +21,7 @@ in
       nixos-install nixos-generate-config nixos-enter;
 
     # Required for --help.
```
```suggestion
```

This comment isn't accurate or helpful anymore.

roberth reviewed (commented):
I like this. They're increasingly like normal packages, which I think is much nicer.
Can't test it now, but otherwise lgtm.

roberth commented:
NixOS isn't quite happy though; see the ofborg error at `nixos/modules/misc/documentation.nix:349:36`

doronbehar commented:
Thanks for the review and the support, I wasn't sure other will support this change. Pushed a change that hopefully will fix the ofborg errors. Will ping again when ofborg is completely green.

doronbehar committed:
nixos/doc: Improve documentation of documentation

- Move contents of README.md from
  nixos/modules/installer/tools/manpages/ to
  nixos/doc/manual/contributing-to-this-manual.chapter.md.
- Don't mention DocBook as its obsolete and too specific.
- Rename derivation attribute name of configuration.nix(5) manual page,
  both on the `contributing-to-this-manual.chapter.md`, and in other
  places.

doronbehar commented:
Pushed an additional commit that improves much the "contributing" documentation.

doronbehar commented:
Tested also the build of the following attributes (ofborg listed and tested them earlier:

```
tests.testers.runNixOSTest-example
tests.nixos-functions.nixos-test
nixos-option
nixos-install-tools
nixos-rebuild
tests.testers.nixosTest-example
tests.trivial-builders.references
home-manager
```

roberth commented:
> Pushed an additional commit that improves much the "contributing" documentation.

Ohh that's laudable but it probably conflicts with @infinisil's broader cleanup
 - https://github.com/NixOS/nixpkgs/pull/245243
 
Not sure what to do about that.

doronbehar commented:
@roberth thanks for the notice. @infinisil's PR seems to be mostly about "how to contribute to docuemtation about *Nixpkgs'*", rather then "how to contribute to docuemtation about NixOS". Hence I'm fairly sure no conflicts will arise as the files he touches there do not intersect with the files changed here.

roberth commented on nixos/doc/manual/contributing-to-this-manual.chapter.md:
```diff
@@ -6,31 +6,104 @@ You can quickly check your edits with the following:
+In any manpage, commands, flags and arguments to the *current* executable should be marked according to their semantics. Commands, flags and arguments passed to *other* executables should not be marked like this and should instead be considered as code examples and marked with `Ql`.
+
+- Use `Fl` to mark flag arguments, `Ar` for their arguments.
+- Repeating arguments should be marked by adding ellipses (`...`).
```
```suggestion
- Repeating arguments should be marked by adding an ellipsis (spelled with periods, `...`).
```

roberth reviewed (approved):
- Done a proofreading. Feel free to commit what you agree with.
- Tested in a VM
  - `man <tool>`
  - `man configuration.nix`
- Confirmed that we don't seem to have html pages for the nixos-* commands yet. Let's not do that in this pr.

:rocket: Can be merged when suggestions have been considered.

doronbehar committed:
nixos/doc: small rephrase in documentation of documentation

Co-authored-by: Robert Hensing <roberth@users.noreply.github.com>

doronbehar commented:
Thanks @roberth for the thorough review! Will give this a few hours today, and will merge afterwards.
